nvtabular/ops/categorify.py

```diff
@@ -196,28 +199,72 @@ def __init__(
         self.stat_name = "categories"
         self.encode_type = encode_type
         self.search_sorted = search_sorted
+        self.hash_type = hash_type
 
         if self.search_sorted and self.freq_threshold:
             raise ValueError(
                 "cannot use search_sorted=True with anything else than the default freq_threshold"
             )
+        if num_buckets:
+            if isinstance(num_buckets, dict):
+                columns = [i for i in num_buckets.keys()]
```

Comment:
```suggestion
                columns = list(num_buckets)
```

Reply:
If the user also passes in a list of columns here - this will get a little weird. Should we warn if columns is not null?

Reply:
@benfred I guess passing a list of columns should not be a case. The user should not send a list of columns. Only 3 options are accepted: dict, tuple or int. Otherwise a TypeError will raise.   
```           
 else:
                raise TypeError(
                    "`num_buckets` must be dict, iterable, or int, got type {}".format(
                        type(num_buckets)
                    )